library/core/src/fmt/num.rs

```diff
@@ -132,27 +119,48 @@ radix! { LowerHex, 16, "0x", x @  0 ..=  9 => b'0' + x, x @ 10 ..= 15 => b'a' +
 radix! { UpperHex, 16, "0x", x @  0 ..=  9 => b'0' + x, x @ 10 ..= 15 => b'A' + (x - 10) }
 
 macro_rules! int_base {
-    (fmt::$Trait:ident for $T:ident as $U:ident -> $Radix:ident) => {
+    (fmt::$Trait:ident for $T:ident -> $Radix:ident) => {
         #[stable(feature = "rust1", since = "1.0.0")]
         impl fmt::$Trait for $T {
             fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                $Radix.fmt_int(*self as $U, f)
+                $Radix.fmt_int(*self, f)
             }
         }
     };
 }
 
 macro_rules! integer {
     ($Int:ident, $Uint:ident) => {
-        int_base! { fmt::Binary   for $Int as $Uint  -> Binary }
-        int_base! { fmt::Octal    for $Int as $Uint  -> Octal }
-        int_base! { fmt::LowerHex for $Int as $Uint  -> LowerHex }
-        int_base! { fmt::UpperHex for $Int as $Uint  -> UpperHex }
-
-        int_base! { fmt::Binary   for $Uint as $Uint -> Binary }
-        int_base! { fmt::Octal    for $Uint as $Uint -> Octal }
-        int_base! { fmt::LowerHex for $Uint as $Uint -> LowerHex }
-        int_base! { fmt::UpperHex for $Uint as $Uint -> UpperHex }
+        int_base! { fmt::Binary   for $Uint -> Binary }
+        int_base! { fmt::Octal    for $Uint -> Octal }
+        int_base! { fmt::LowerHex for $Uint -> LowerHex }
+        int_base! { fmt::UpperHex for $Uint -> UpperHex }
+
+        // Format signed integers as unsigned (two’s complement representation).
+        #[stable(feature = "rust1", since = "1.0.0")]
+        impl fmt::Binary for $Int {
+            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                fmt::Binary::fmt(&(*self as $Uint), f)
```

Comment:
Mentioned above but we now have `.cast_unsigned()` which is slightly cleaner than `as $Uint`